sdk/react/collapsingBarUtils.js

```diff
@@ -1,79 +1,101 @@
 'use strict';
 
-var CollapsingBarUtils = {
+// MARK: - Constants
+
+const constants = {
+  moreOptions: 'moreOptions',
+  moveToMoreOptions: 'moveToMoreOptions',
+  controlBar: 'controlBar',
+  keep: 'keep'
+};
+
+const CollapsingBarUtils = {
 
   // @param barWidth numeric.
   // @param orderedItems array of left to right ordered items. Each item meets the skin's "button" schema.
   // @return {fit:[items that fit in the barWidth], overflow:[items that did not fit]}.
   // Note: items which do not meet the item spec will be removed and not appear in the results.
-  collapse: function( barWidth, orderedItems ) {
-    if( isNaN( barWidth ) || barWidth === undefined ) { return { fit : orderedItems, overflow : [] };; }
-    if( ! orderedItems ) { return []; }
-    var self = this;
-    var validItems = orderedItems.filter( function(item) { return self._isValid(item); } );
-    var r = this._collapse( barWidth, validItems );  
-    return r;
+  collapse: function(barWidth, orderedItems) {
```

Comment:
Should we use inline functions?
i.e.
```
collapse = (barWidth, orderedItems) => { ... } 
```

Reply:
if we will use arrow functions a js context will want to use global scope and it will can't find functions